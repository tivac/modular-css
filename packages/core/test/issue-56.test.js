"use strict";

var dedent = require("dedent"),
    namer  = require("test-utils/namer.js"),
    
    Processor = require("../processor.js");

describe("/issues", function() {
    describe("/56", function() {
        it("should prune rules that only compose, but leave them in the exports", function() {
            var processor = new Processor({
                    namer
                });
            
            return processor.string(
                    "./packages/core/test/specimens/issues/56.css",
                    dedent(`
                        .booga { color: red }
                        .fooga { composes: booga }
                        .fooga:hover { color: blue }
                        .wooga { composes: booga }
                    `)
            )
            .then((result) => {
                expect(result.exports).toMatchSnapshot();

                return processor.output();
            })
            .then((result) => expect(result.css).toMatchSnapshot());
        });
    });
});
