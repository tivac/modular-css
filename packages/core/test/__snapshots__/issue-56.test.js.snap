// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/issues /56 should prune rules that only compose, but leave them in the exports 1`] = `
Object {
  "booga": Array [
    "booga",
  ],
  "fooga": Array [
    "booga",
    "fooga",
  ],
  "wooga": Array [
    "booga",
    "wooga",
  ],
}
`;

exports[`/issues /56 should prune rules that only compose, but leave them in the exports 2`] = `
"/* packages/core/test/specimens/issues/56.css */
.booga {
    color: red
}
.fooga:hover {
    color: blue
}"
`;
